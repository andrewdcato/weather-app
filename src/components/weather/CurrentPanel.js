import { getCardinalDirection, getMoonPhase, getTime } from '../../helpers';

function ConditionCard({ description = null, value}) {
  return (
    <>
      <div className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40">
        <div className="text-left">
          {!!description && <span className="font-semibold">{description}</span>} {value}
        </div>
      </div>
    </>
  )
}

export default function CurrentPanel({
  current, today, date: { month, day, year, dayOfWeek}
}) {
  return (
    <>
      <div className="flex mb-4 justify-between items-center">
        <div className="flex flex-col">
          <h5 className="mb-0 font-medium text-3xl">{dayOfWeek}</h5>
          <h6>{month} {day}, {year}</h6>
        </div>
        <div className="flex flex-col text-right">
          <h5 className="font-medium text-3xl"><span>{parseInt(current?.temp)}&deg; F</span></h5>
          <h6>{current?.weather[0].main}</h6>
        </div>
      </div>

      <h3 className="text-xl">Current Conditions</h3>
      <div className="mt-6">
        <div className="grid xs:grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-2">
          <ConditionCard description="Feels Like" value={
            <span>{parseInt(current?.feels_like)}&deg; F</span>
          }/>
          <ConditionCard description="Pressure" value={
            <span>{current?.pressure} hPa</span>
          } />
          <ConditionCard description="Sunrise" value={
            <span>{getTime(current?.sunrise)}</span>
          } />
          <ConditionCard description="Precipitation" value={
            <span>{parseInt(today?.pop * 100)}%</span>
          } />
          <ConditionCard description="Humidity" value={
            <span>{current?.humidity}%</span>
          } />
          <ConditionCard description="Sunset" value={
            <span>{getTime(current?.sunset)}</span>
          } />
          <ConditionCard description="Wind" value={
            <span>{parseInt(current?.wind_speed)} MPH {getCardinalDirection(current?.wind_deg)}</span>
          } />
          <ConditionCard description="UV Index" value={
            <span>{current?.uvi}</span>
          } />
          <ConditionCard value={
            <span>{getMoonPhase(today?.moon_phase || 0)}</span>
          } />
        </div>
      </div>
    </>
  )
}
