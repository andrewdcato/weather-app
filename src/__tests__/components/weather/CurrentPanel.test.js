import React from 'react';
import renderer from 'react-test-renderer';
import CurrentPanel from '../../../components/weather/CurrentPanel';

test('properly renders with no forecast data', () => {
  const current = renderer.create(
    <CurrentPanel date={global.TEST_DATE} />
  ).toJSON();

  expect(current).toMatchSnapshot();
});

test('properly renders with forecast data', () => {
  const current = renderer.create(
    <CurrentPanel
      current={global.TEST_CURRENT_FORECAST}
      date={global.TEST_DATE}
      today={global.TEST_TODAY}
    />
  ).toJSON();

  expect(current).toMatchSnapshot();
});

