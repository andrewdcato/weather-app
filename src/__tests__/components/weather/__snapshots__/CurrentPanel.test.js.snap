// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`properly renders with forecast data 1`] = `
Array [
  <div
    className="flex mb-4 justify-between items-center"
  >
    <div
      className="flex flex-col"
    >
      <h5
        className="mb-0 font-medium text-3xl"
      >
        Wednesday
      </h5>
      <h6>
        April
         
        13
        , 
        2016
      </h6>
    </div>
    <div
      className="flex flex-col text-right"
    >
      <h5
        className="font-medium text-3xl"
      >
        <span>
          62
          째 F
        </span>
      </h5>
      <h6>
        Clouds
      </h6>
    </div>
  </div>,
  <h3
    className="text-xl"
  >
    Current Conditions
  </h3>,
  <div
    className="mt-6"
  >
    <div
      className="grid xs:grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-2"
    >
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
          <span
            className="font-semibold"
          >
            Feels Like
          </span>
           
          <span>
            61
            째 F
          </span>
        </div>
      </div>
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
          <span
            className="font-semibold"
          >
            Pressure
          </span>
           
          <span>
            1019
             hPa
          </span>
        </div>
      </div>
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
          <span
            className="font-semibold"
          >
            Sunrise
          </span>
           
          <span>
            7 am
          </span>
        </div>
      </div>
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
          <span
            className="font-semibold"
          >
            Precipitation
          </span>
           
          <span>
            0
            %
          </span>
        </div>
      </div>
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
          <span
            className="font-semibold"
          >
            Humidity
          </span>
           
          <span>
            77
            %
          </span>
        </div>
      </div>
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
          <span
            className="font-semibold"
          >
            Sunset
          </span>
           
          <span>
            4 pm
          </span>
        </div>
      </div>
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
          <span
            className="font-semibold"
          >
            Wind
          </span>
           
          <span>
            14
             MPH 
            S
          </span>
        </div>
      </div>
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
          <span
            className="font-semibold"
          >
            UV Index
          </span>
           
          <span>
            0
          </span>
        </div>
      </div>
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
           
          <span>
            Waxing Gibbous
          </span>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`properly renders with no forecast data 1`] = `
Array [
  <div
    className="flex mb-4 justify-between items-center"
  >
    <div
      className="flex flex-col"
    >
      <h5
        className="mb-0 font-medium text-3xl"
      >
        Wednesday
      </h5>
      <h6>
        April
         
        13
        , 
        2016
      </h6>
    </div>
    <div
      className="flex flex-col text-right"
    >
      <h5
        className="font-medium text-3xl"
      >
        <span>
          NaN
          째 F
        </span>
      </h5>
      <h6 />
    </div>
  </div>,
  <h3
    className="text-xl"
  >
    Current Conditions
  </h3>,
  <div
    className="mt-6"
  >
    <div
      className="grid xs:grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-2"
    >
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
          <span
            className="font-semibold"
          >
            Feels Like
          </span>
           
          <span>
            NaN
            째 F
          </span>
        </div>
      </div>
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
          <span
            className="font-semibold"
          >
            Pressure
          </span>
           
          <span>
             hPa
          </span>
        </div>
      </div>
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
          <span
            className="font-semibold"
          >
            Sunrise
          </span>
           
          <span>
            Invalid Date
          </span>
        </div>
      </div>
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
          <span
            className="font-semibold"
          >
            Precipitation
          </span>
           
          <span>
            NaN
            %
          </span>
        </div>
      </div>
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
          <span
            className="font-semibold"
          >
            Humidity
          </span>
           
          <span>
            %
          </span>
        </div>
      </div>
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
          <span
            className="font-semibold"
          >
            Sunset
          </span>
           
          <span>
            Invalid Date
          </span>
        </div>
      </div>
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
          <span
            className="font-semibold"
          >
            Wind
          </span>
           
          <span>
            NaN
             MPH 
          </span>
        </div>
      </div>
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
          <span
            className="font-semibold"
          >
            UV Index
          </span>
           
          <span />
        </div>
      </div>
      <div
        className="px-3 mx-3 flex flex-col rounded shadow bg-blue-200 bg-opacity-40"
      >
        <div
          className="text-left"
        >
           
          <span>
            New Moon
          </span>
        </div>
      </div>
    </div>
  </div>,
]
`;
